src/evaluators/CallExpression.js

```diff
@@ -66,17 +66,29 @@ function callBothFunctionsAndJoinTheirEffects(
   invariant(Value.isTypeCompatibleWith(func1.getType(), FunctionValue));
   invariant(Value.isTypeCompatibleWith(func2.getType(), FunctionValue));
 
-  let [compl1, gen1, bindings1, properties1, createdObj1] = realm.evaluateForEffects(
+  let {
+    result: compl1,
```

Comment:
It will probably be quite nice to just stick with the property names of Effects.

Reply:
What would you say is more readable, to keep destructuring:
```js
let {
    result: result1,
    generator: generator1,
    modifiedBindings: modifiedBindings1,
    modifiedProperties: modifiedProperties1,
    createdObjects: createdObjects1,
  } = realm.evaluateForEffects(...
```
or to name the objects and call their properties directly ?
```js
const e1 = realm.evaluateForEffects(...)
...
new Effects(e1.result, e1.generator, e1.modifiedBindings, e1.modifiedProperties, e1.createdObjects),
...
```

Reply:
I went for the second option whenever possible, as it is a bit more terse.